pkg/commands/pull_request.go

```diff
@@ -174,6 +185,18 @@ func getRepoInfoFromURL(url string) *RepoInformation {
 		}
 	}
 
+	// there has to be a "nicer" way
+	if strings.HasPrefix(url, "ssh://git@stash") {
```

Comment:
I've spent some time thinking about this and I think there's two things to note here:
1) extracting the owner/repo from the URL is difficult when done with string methods
2) different git hosting sites will have different URL structures.

I think we should change this method to use regexes e.g.:
```go
func getRepoInfoFromURL(url string) *RepoInformation {
	regexStr := []string{
		`^https?://.*/(?P<owner>.*)/(?P<repo>.*?)(\.git)?$`,
		`^git@.*:(?P<owner>.*)/(?P<repo>.*?)(\.git)?$`,
	}

	for _, regexStr := range regexStr {
		re := regexp.MustCompile(regexStr)
		matches := utils.FindNamedMatches(re, url)
		if matches != nil {
			return &RepoInformation{
				Owner:      matches["owner"],
				Repository: matches["repo"],
			}
		}
	}

	return nil
}
```
And we can add a new file to the utils package called 'regexp.go' which defines:
```go
func FindNamedMatches(regex *regexp.Regexp, str string) map[string]string {
	match := regex.FindStringSubmatch(str)

	if len(match) == 0 {
		return nil
	}

	results := map[string]string{}
	for i, value := range match {
		results[regex.SubexpNames()[i]] = value
	}
	return results
}
```

Then the next step would be to define a slice of regexes on each git service so that we're not mixing specific with general logic e.g.:
```go
var defaultUrlRegexes = []string{
	`^https?://.*/(?P<owner>.*)/(?P<repo>.*?)(\.git)?$`,
	`^git@.*:(?P<owner>.*)/(?P<repo>.*?)(\.git)?$`,
}

...

case "bitbucket":
	service = &Service{
		Name: repositoryDomain,
		pullRequestURLIntoDefaultBranch: func(owner string, repository string, from string) string {
			return fmt.Sprintf("https://%s/%s/%s/pull-requests/new?source=%s&t=1", siteDomain, owner, repository, from)
		},
		pullRequestURLIntoTargetBranch: func(owner string, repository string, from string, to string) string {
			return fmt.Sprintf("https://%s/%s/%s/pull-requests/new?source=%s&dest=%s&t=1", siteDomain, owner, repository, from, to)
		},
		urlRegexes: defaultUrlRegexes,
	}
```

And for stash we can use our own custom regexes instead of the default which each other service uses.

We would then want to add a couple tests for the stash service in `pkg/commands/pull_request_test.go`

Lemme know your thoughts :)